frontend/src/metabase/static-viz/components/XYChart/utils/series.ts

```diff
@@ -50,16 +50,38 @@ export const sortSeries = (series: Series[], type: XAxisType) => {
   });
 };
 
-export const calculateStackedItems = (series: Series[]) => {
+export const calculateStackedItems = (multipleSeries: Series[]) => {
+  const dimensionSeriesIndexMap = multipleSeries.reduce(
+    (map, series, seriesIndex) => {
+      series.data.forEach(datum => {
+        const dimension = getX(datum);
+        if (!map[dimension]) {
+          map[dimension] = {};
+        }
+        map[dimension][seriesIndex] = datum;
+      });
+      return map;
+    },
+    {} as {
+      [dimension: string]: {
+        [seriesIndex: number]: SeriesDatum;
+      };
+    },
+  );
+
   // Stacked charts work only for a single dataset with one dimension
-  return series.map((s, seriesIndex) => {
-    const stackedData = s.data.map((datum, datumIndex) => {
+  return multipleSeries.map((series, seriesIndex) => {
```

Comment:
In static viz, we name the variable that represent multiple series `multipleSeries`, and the variable that represent a single series `series`

e.g. https://github.com/metabase/metabase/blob/c05b859a103976fbb6512e21450da5f469fdd788/frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx#L350